src/components/CheckBoxFilterContainer.ts

```diff
@@ -0,0 +1,203 @@
+            const filterBy = isChecked ? this.props.filterBy : this.props.unCheckedFilterBy;
+            const constraint = isChecked ? this.props.constraint : this.props.unCheckedConstraint;
+            const attributeValue = isChecked ? this.props.attributeValue : this.props.unCheckedAttributeValue;
+
+            if (filterBy === "XPath") {
+                return constraint.indexOf(`[%CurrentObject%]'`) !== -1
+                    ? constraint.replace(`'[%CurrentObject%]'`, this.props.mxObject.getGuid())
+                    : constraint;
+            } else if (filterBy === "attribute") {
+                return this.getAttributeConstaint(attribute, attributeValue);
+            } else {
+               return "";
+            }
+        }
+    }
+
+    private getAttributeConstaint(attribute: string, attributeValue: string): string | HybridConstraint {
+        const { targetListView } = this.state;
+
+        if (window.device) {
+            const constraints: HybridConstraint = {
+                attribute,
+                operator: "contains",
+                path: this.props.listViewEntity,
+                value: attributeValue
+            };
+            return constraints;
+        }
+        if (targetListView && targetListView._datasource) {
+            const mxObject = targetListView._datasource._pageObjs[0];
+            if (mxObject && mxObject.isEnum(attribute)) {
+                return `[${attribute}='${attributeValue.trim()}']`;
+            } else if (mxObject && mxObject.isBoolean(attribute)) {
+                return `[${attribute} = ${attributeValue.trim().toLowerCase()}()]`;
+            } else {
+                return `[contains(${attribute},'${attributeValue}')]`;
+            }
+        }
+    }
+
+    private connectToListView() {
+        const filterNode = findDOMNode(this).parentNode as HTMLElement;
+        const targetNode = Utils.findTargetNode(filterNode);
+        let targetListView: ListView | null = null;
+
+        if (targetNode) {
+            targetListView = dijitRegistry.byNode(targetNode);
+            if (targetListView) {
+                if (!targetListView.__customWidgetDataSourceHelper) {
+                    try {
+                        targetListView.__customWidgetDataSourceHelper = new DataSourceHelper(targetListView);
+                    } catch (error) {
+                        console.log("failed to instantiate DataSourceHelper \n" + error); // tslint:disable-line
+                    }
+                } else if (!DataSourceHelper.checkVersionCompatible(targetListView.__customWidgetDataSourceHelper.version)) {
+                    console.log("Compartibility issue"); // tslint:disable-line
+                }
+                this.dataSourceHelper = targetListView.__customWidgetDataSourceHelper;
+                this.setState({
+                    listViewAvailable: !!targetListView,
```

Comment:
Why is the state `listViewAvailable`, when there is are error... that should not happen when there is an error  with data source helepr

Reply:
Done